sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashOuterJoin.scala

```diff
@@ -76,8 +101,12 @@ override def outputPartitioning: Partitioning = joinType match {
       rightIter: Iterable[InternalRow]): Iterator[InternalRow] = {
     val ret: Iterable[InternalRow] = {
       if (!key.anyNull) {
-        val temp = rightIter.collect {
-          case r if boundCondition(joinedRow.withRight(r)) => joinedRow.copy()
+        val temp = if (rightIter != null) {
+          rightIter.collect {
+            case r if boundCondition(joinedRow.withRight(r)) => joinedRow.copy()
+          }
+        } else {
+          List()
```

Comment:
I think that you can use `List.empty` here, which, as far as I know, returns an immutable singleton.  Not sure if `List()` creates a new instance or not...